snow/engine/snowman/transitive.go

```diff
@@ -515,22 +515,26 @@ func (t *Transitive) Start(ctx context.Context, startReqID uint32) error {
 	if oracleBlk, ok := lastAccepted.(snowman.OracleBlock); ok {
 		options, err := oracleBlk.Options(ctx)
 		switch {
-		case err == snowman.ErrNotOracle:
-			// if there aren't blocks we need to deliver on startup, we need to set
-			// the preference to the last accepted block
-			if err := t.VM.SetPreference(ctx, lastAcceptedID); err != nil {
-				return err
-			}
-		case err != nil:
-			return err
-		default:
+		case err == nil:
 			issuedMetric := t.metrics.issued.WithLabelValues(builtSource)
 			for _, blk := range options {
 				// note that deliver will set the VM's preference
 				if err := t.deliver(ctx, t.Ctx.NodeID, blk, false, issuedMetric); err != nil {
 					return err
 				}
 			}
+		case err != snowman.ErrNotOracle:
+			t.Ctx.Log.Warn("failed to set preferences for oracle block",
+				zap.Stringer("blkID", lastAcceptedID),
+				zap.Error(err),
+			)
```

Comment:
This is logged as a `Warn` rather than `Debug` because we have already accepted the proposal block at this point.

It might make sense to elevate this to `Error` - as if everyone in the network observed this error at this block, the chain may stall.